python/dlisio/lis/load.py

```diff
@@ -96,52 +144,85 @@ def read_as_tapemark(f):
         first = index.explicits()[0] if len(index.explicits()) else None
         if is_delimiter(first):
             try:
-                record = f.read_record(first)
-                f.close()
-            except Exception as e:
-                # This is very unlikely to happen as index_records has already
-                # done the sanity checking needed for succesfully reading the
-                # record.
-                # However read_records can in theory still fail on blocked IO
-                # and other non-LIS related issues.
-                msg =  'dlisio.lis.load: Could not read record {}, indexing stopped\n'
-                msg += 'Reason: {}\nFilepath: {}'
-                logging.error(msg.format(first, e, path))
-                f.close()
-                break
-
-            if first.type == core.lis_rectype.reel_header:
-                reel = HeaderTrailer(record)
-
-            elif first.type == core.lis_rectype.reel_trailer:
-                # The reel is already assigned to the LF's at this point, so
-                # just assign the trailer to that instance before initiating a
-                # new instance for the next reel.
-                reel.rawtrailer = record
-                reel = HeaderTrailer()
-
-            elif first.type == core.lis_rectype.tape_header:
-                tape = HeaderTrailer(record)
-
-            elif first.type == core.lis_rectype.tape_trailer:
-                # The tape is already assigned to the LF's at this point, so
-                # just assign the trailer to that instance before initiating a
-                # new instance for the next tape.
-                tape.rawtrailer = record
-                tape = HeaderTrailer()
-
+                if index.size() != 1:
```

Comment:
Style-comment that you can ignore. Maybe this becomes a bit more readable if we do:
```
try:
    if index.size() != 1:
        self.error_hander.log( ... )
 finally:
     self.file.close()
     
 try:
    record = self.file.read_record(first)
 except Exception as e:
     self.error_hander.log( ... )
 finally:
     self.file.close()